### ISSUE TITLE
Can iptsd work on non linux-surface kernels?

### ISSUE THREAD

**Issue by 1900011604** [linux-surface/linux-surface#515]
My kernel is x86_64 Linux 5.13.4-1-MANJARO, it supports surface batteries now, but no touchscreen. Can I install and use iptsd without changing the kernel?

**Comment by StollD** [comment]
No, because while the SAM drivers (for battery etc.) have been upstreamed, the driver for the touchscreen is not yet upstream. iptsd is just the component that processes a raw heatmap into touch coordinates. Starting the hardware, controlling it and sending its data to iptsd still needs to be done by a kernel driver.

If you dont want to install a patched kernel, you could alternatively install the ipts kernel driver (https://github.com/linux-surface/intel-precise-touch) with DKMS, but then you will also miss some of the other patches that we carry in the surface kernel, like the fixes for wifi and the camera support.

**Comment by SamLukeYes** [comment]
You can try out my AUR package [ipts-uapi-dkms-git](https://aur.archlinux.org/packages/ipts-uapi-dkms-git/)
